spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DefaultEnvironmentPostProcessor.java

```diff
@@ -35,39 +35,53 @@
 import org.springframework.core.io.Resource;
 
 /**
- * Contributes the values from {@code dataflow-server.yml} if it exists, before any of Spring Boot's normal
+ * Contributes the values from {@code META-INF/dataflow-server-defaults.yml} and
+ * {@code dataflow-server.yml} if it exists, before any of Spring Boot's normal
  * configuration contributions apply. This has the effect of supplying overridable defaults to the
  * various Spring Cloud Data Flow Deployer SPI implementations that in turn override the defaults
  * provided by Spring Boot.
  *
  * @author Josh Long
+ * @author Janne Valkealahti
  */
 public class DefaultEnvironmentPostProcessor implements EnvironmentPostProcessor, Ordered {
 
-	private Resource resource = new ClassPathResource("/dataflow-server.yml");
+	private static Log logger = LogFactory.getLog(DefaultEnvironmentPostProcessor.class);
+	private final Resource serverResource = new ClassPathResource("/dataflow-server.yml");
+	private final Resource serverDefaultsResource = new ClassPathResource("META-INF/dataflow-server-defaults.yml");
 
 	@Override
 	public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application) {
+		Map<String, Object> internalDefaults = new HashMap<>();
 		Map<String, Object> defaults = new HashMap<>();
 		MutablePropertySources existingPropertySources = environment.getPropertySources();
 
-		this.contributeDefaults(defaults);
+		contributeDefaults(internalDefaults, serverDefaultsResource);
+		contributeDefaults(defaults, serverResource);
 
 		String defaultPropertiesKey = "defaultProperties";
 
 		if (!existingPropertySources.contains(defaultPropertiesKey) ||
 				existingPropertySources.get(defaultPropertiesKey) == null) {
+			existingPropertySources.addLast(new MapPropertySource(defaultPropertiesKey, internalDefaults));
 			existingPropertySources.addLast(new MapPropertySource(defaultPropertiesKey, defaults));
 		}
 		else {
 			PropertySource<?> propertySource = existingPropertySources.get(defaultPropertiesKey);
 			@SuppressWarnings("unchecked")
 			Map<String, Object> mapOfProperties = Map.class.cast(propertySource.getSource());
+			for (String k : internalDefaults.keySet()) {
+				Set<String> setOfPropertyKeys = mapOfProperties.keySet();
+				if (!setOfPropertyKeys.contains(k)) {
+					mapOfProperties.put(k, internalDefaults.get(k));
+					logger.info(k + '=' + internalDefaults.get(k));
```

Comment:
Yea, though we should be able to see the properties via the actuator endpoint if needed...so while not related to this PR, I think I'll change it to debug on merge.